instrumentation/okhttp/okhttp-2.2/src/test/groovy/OkHttp2AsyncTest.groovy

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import com.squareup.okhttp.Callback
+import com.squareup.okhttp.Headers
+import com.squareup.okhttp.MediaType
+import com.squareup.okhttp.Request
+import com.squareup.okhttp.RequestBody
+import com.squareup.okhttp.Response
+import com.squareup.okhttp.internal.http.HttpMethod
+
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.atomic.AtomicReference
+
+import static java.util.concurrent.TimeUnit.SECONDS
+
+class OkHttp2AsyncTest extends OkHttp2Test {
+  @Override
+  int doRequest(String method, URI uri, Map<String, String> headers, Closure callback) {
+    def body = HttpMethod.requiresRequestBody(method) ? RequestBody.create(MediaType.parse("text/plain"), "") : null
+    def request = new Request.Builder()
+      .url(uri.toURL())
+      .method(method, body)
+      .headers(Headers.of(HeadersUtil.headersToArray(headers)))
+      .build()
+
+    AtomicReference<Response> responseRef = new AtomicReference()
```

Comment:
Maybe easier to use one `CompletableFuture` instead of two `AtomicReference` and a `CountDownLatch`? Callback will call either `complete` or `completeExceptionally` and test can end in just one line `return future.get(10, SECONDS)`.

Reply:
Ah, this didn't end up working out because the Java 7 tests failed, oh well, it was a nice idea 👍